js/Keyboard.js

```diff
@@ -0,0 +1,232 @@
+      } else if (this.isCaps) {
+        if (this.shiftKey) {
+          this.printOutput(keyObj, keyObj.sub.innerHTML ? keyObj.shift : keyObj.small);
+        } else {
+          this.printOutput(keyObj, !keyObj.sub.innerHTML ? keyObj.shift : keyObj.small);
+        }
+      }
+    } else if (type.match(/keyup|mouseup/)) {
+      if (code.match(/Shift/)) {
+        this.shiftKey = false;
+        this.switchUpperCase(false);
+      }
+      if (code.match(/Control/)) this.ctrlKey = false;
+      if (code.match(/Meta/)) this.commandKey = false;
+      if (!code.match(/Caps/)) keyObj.div.classList.remove('active');
+    }
+  }
+
+  switchLanguage = () => {
+    const langAbbr = Object.keys(language); // ['en', 'ru']
+    let langIdx = langAbbr.indexOf(this.container.dataset.language);
+    this.keyBase = langIdx + 1 < langAbbr.length ? language[langAbbr[langIdx += 1]]
+      : language[langAbbr[langIdx -= langIdx]];
+
+    this.container.dataset.language = langAbbr[langIdx];
+    storage.set('kbLang', langAbbr[langIdx]);
+
+    this.keyButtons.forEach((button) => {
+      const keyObj = this.keyBase.find((key) => key.code === button.code);
+      if (!keyObj) return;
+      button.shift = keyObj.shift;
+      button.small = keyObj.small;
+      if (keyObj.shift && keyObj.shift.match(/[^a-zA-Zа-яА-ЯёЁ0-9]/g)) {
+        button.sub.innerHTML = keyObj.shift;
+      } else {
+        button.sub.innerHTML = '';
+      }
+      button.letter.innerHTML = keyObj.small;
+    });
+
+    if (this.isCaps) this.switchUpperCase(true);
+  }
+
+  switchUpperCase = (isTrue) => {
+    if (isTrue) {
+      this.keyButtons.forEach((button) => {
+        if (button.sub.innerHTML) {
+          if (this.shiftKey) {
+            button.sub.classList.add('sub-active');
+            button.letter.classList.add('sub-inactive');
+          }
+        }
+        if (!button.isFnKey && this.isCaps && !this.shiftKey && !button.sub.innerHTML) {
+          button.letter.innerHTML = button.shift;
+        } else if (!button.isFnKey && this.isCaps && this.shiftKey) {
+          button.letter.innerHTML = button.small;
+        } else if (!button.isFnKey && !button.sub.innerHTML) {
+          button.letter.innerHTML = button.shift;
+        }
+      });
+    } else {
+      this.keyButtons.forEach((button) => {
+        if (button.sub.innerHTML && !button.isFnKey) {
+          button.sub.classList.remove('sub-active');
+          button.letter.classList.remove('sub-inactive');
+
+          if (this.isCaps && this.shiftKey) {
+            button.letter.innerHTML = button.small;
+          } else if (this.isCaps && !this.shiftKey) {
+            button.letter.innerHTML = button.shift;
+          }
+        } else if (!button.isFnKey && !button.sub.innerHTML) {
+          if (this.isCaps) {
+            button.letter.innerHTML = button.shift;
+          } else {
+            button.letter.innerHTML = button.small;
+          }
+        }
```

Comment:
Finally found something :)
It is better not to produce such code, because:
- very hard to read, with every else / if readability is getting worse.

To avoid such stuff in the future,
 **always try to make things modular --- split them** 
Do not afraid to create additional function with meaningful names etc.
Also take a look here for example, just to try to avoid this in the future:
https://refactoring.guru/replace-nested-conditional-with-guard-clauses
https://refactoring.guru/decompose-conditional